store/src/java/com/zimbra/cs/milter/MilterHandler.java

```diff
@@ -354,6 +355,23 @@ private void SMFIC_Rcpt() throws ServiceException {
             connection.send(new MilterPacket(SMFIR_TEMPFAIL));
             return;
         }
+        // ZBUG-2165: prevent forwarding to non-permitted distribution list
+        final Account account = prov.getAccountByName(rcpt);
+        if (account != null && account.getPrefMailForwardingAddress() != null) {
+            ZimbraLog.milter.trace("account: %s", account.getName());
+            for (final String a : account.getPrefMailForwardingAddress().split(",")) {
+                final String address = a.trim();
+                ZimbraLog.milter.trace("address: %s", address);
+                if (!address.isEmpty() && prov.isDistributionList(address)) {
+                    final Group group = prov.getGroupBasic(Key.DistributionListBy.name, address);
+                    if (group != null && !accessMgr.canDo(account, group, User.R_sendToDistList, false)) {
+                        ZimbraLog.milter.debug("Recipient %s is not allowed to forward to this distribution list: %s", account.getName(), group.getName());
+                        SMFIR_ReplyCode("571", "571 Recipient " + account.getName() + " is not allowed to forward to this distribution list: " + group.getName());
+                        return;
+                    }
+                }
+            }
```

Comment:
@wilfredo-alfaro You have get rid of `try` `catch` blocks but can you please address @desouzas concern.